pygmt/datatypes/grid.py

```diff
@@ -102,4 +144,224 @@ class _GMT_GRID_HEADER(ctp.Structure):  # noqa: N801
+        >>> da.coords["lon"]  # doctest: +NORMALIZE_WHITESPACE, +ELLIPSIS
+        <xarray.DataArray 'lon' (lon: 8)>...
+        array([-54.5, -53.5, -52.5, -51.5, -50.5, -49.5, -48.5, -47.5])
+        Coordinates:
+          * lon      (lon) float64... -54.5 -53.5 -52.5 -51.5 -50.5 -49.5 -48.5 -47.5
+        Attributes:
+            long_name:      longitude
+            units:          degrees_east
+            standard_name:  longitude
+            axis:           X
+            actual_range:   [-55. -47.]
+        >>> da.coords["lat"]  # doctest: +NORMALIZE_WHITESPACE, +ELLIPSIS
+        <xarray.DataArray 'lat' (lat: 14)>...
+        array([-23.5, -22.5, -21.5, -20.5, -19.5, -18.5, -17.5, -16.5, -15.5, -14.5,
+            -13.5, -12.5, -11.5, -10.5])
+        Coordinates:
+          * lat      (lat) float64... -23.5 -22.5 -21.5 -20.5 ... -12.5 -11.5 -10.5
+        Attributes:
+            long_name:      latitude
+            units:          degrees_north
+            standard_name:  latitude
+            axis:           Y
+            actual_range:   [-24. -10.]
+        >>> da.gmt.registration, da.gmt.gtype
+        (1, 1)
+        """
+        # The grid header
+        header = self.header.contents
+
+        # The data array without paddings
+        pad = header.pad[:]
+        data = np.reshape(self.data[: header.mx * header.my], (header.my, header.mx))[
+            pad[2] : header.my - pad[3], pad[0] : header.mx - pad[1]
+        ]
+
+        # Default dimension names and grid name
+        dims, name = ("y", "x"), "z"
+
+        # The x and y coordinates
+        coords = {
+            "x": self.x[: header.n_columns],
+            "y": self.y[: header.n_rows],
+        }
+
+        # Attributes for the coordinates and the data (i.e., gloabal attributes).
+        # Refer to the 'gmtnc_put_units' and 'gmtnc_get_units' functions in 'gmt_nc.c'.
+        attrs: dict = {"x": {}, "y": {}, "z": {}}
+        newdims = {"x": "x", "y": "y"}  # Map default dims to new dims
+        for dim, nameunits in (
+            ("x", header.x_units),
+            ("y", header.y_units),
+            ("z", header.z_units),
+        ):
+            long_name, units = _parse_nameunits(nameunits.decode())
+            if long_name:
+                attrs[dim]["long_name"] = long_name
+            if units:
+                attrs[dim]["units"] = units
+                if units == "degrees_east":
+                    attrs[dim]["standard_name"] = "longitude"
+                    newdims[dim] = "lon"
+                elif units == "degrees_north":
+                    attrs[dim]["standard_name"] = "latitude"
+                    newdims[dim] = "lat"
+            if dim == "z":
+                attrs[dim]["actual_range"] = np.array([header.z_min, header.z_max])
+            else:
+                attrs[dim]["axis"] = dim.upper()
+                idx = 0 if dim == "x" else 2
+                attrs[dim]["actual_range"] = np.array(header.wesn[idx : idx + 2])
```

Comment:
Could reuse some of this logic (and `_parse_nameunits`) in wrapping `GMT_IMAGE` at #3128, could we make a `_get_attrs` function out of this code block, and cherry-pick this into another PR? Or just the `_parse_nameunits` function will do.

Reply:
OK, I've made the `_parse_header` function which parses the grid header to get grid metadata that will be used for creating an xarray.DataArray object. 

The PR is at https://github.com/GenericMappingTools/pygmt/pull/3134.